parser/prism/Factory.cc

```diff
@@ -0,0 +1,512 @@
+#include "parser/prism/Factory.h"
+#include "parser/prism/Helpers.h"
+#include "parser/prism/Parser.h"
+
+namespace sorbet::parser::Prism {
+
+using namespace std;
+using namespace std::literals::string_view_literals;
+
+// See Prism's `include/prism/defines.h`
+inline constexpr auto prismFree = [](void *p) { xfree(p); };
+
+// A unique_ptr to memory that will be owned by Prism.
+// * Must be allocated from one of the Prism allocator functions
+//   See Prism's `include/prism/defines.h`
+// * Ownership can be transfered to a Prism-owned AST via `release()`.
+// * In the event of an exception, will be freed correctly using Prism's `xfree()`.
+template <typename T> using PrismUniquePtr = std::unique_ptr<T, decltype(prismFree)>;
+
+pm_node_list_t Factory::copyNodesToList(const vector<pm_node_t *> &nodes) const {
```

Comment:
Since this function doesn't really care about the vector-ness of the list, it should take an `absl::Span<pm_node_t *>`, so the caller isn't forced to allocate a vector.